StreamChatClient_v3/ChatClient.swift

```diff
@@ -42,27 +42,21 @@ public typealias ChatClient = Client<DefaultDataTypes>
 /// If you don't need to specify your custom extra data types for `User`, `Channel`, or `Message`, use the convenient non-generic
 /// typealias `ChatClient` which specifies the default extra data types.
 public class Client<ExtraData: ExtraDataTypes> {
-    /// An object containing all dependencies of `Client`
-    struct Environment {
-        var apiClientBuilder: (
-            _ apiKey: APIKey,
-            _ baseURL: URL,
-            _ sessionConfiguration: URLSessionConfiguration
-        ) -> APIClient = { APIClient(apiKey: $0, baseURL: $1, sessionConfiguration: $2) }
-        
-        var webSocketClientBuilder: (
-            _ urlRequest: URLRequest,
-            _ sessionConfiguration: URLSessionConfiguration,
-            _ eventDecoder: AnyEventDecoder,
-            _ eventMiddlewares: [EventMiddleware]
-        ) -> WebSocketClient = { WebSocketClient(urlRequest: $0, sessionConfiguration: $1, eventDecoder: $2, eventMiddlewares: $3) }
-        
-        var databaseContainerBuilder: (_ kind: DatabaseContainer.Kind) throws
-            -> DatabaseContainer = { try DatabaseContainer(kind: $0) }
-    }
+    /// The id of the currently logged in user.
+    @Atomic public var currentUserId: UserId = .anonymous
     
-    /// The currently logged-in user.
-    public let currentUser: UserModel<ExtraData.User>
+    /// The currently logged-in user data.
+    ///
+    /// Returns `nil` if the user is not fully logged-in yet. Always wait for the `setUser` completion block before
+    /// accessing this value.
+    public var currentUser: CurrentUserModel<ExtraData.User>? {
```

Comment:
1. It should be `internal(set)`
2. I think it shouldn't be optional, because we have anonymous by default as placeholder. It will simplify a lot of our lives.

Reply:
1. It's a computed variable, you can't mutate it.

Reply:
2. I see what you mean but I think this will be a smaller issue in v3. Since the solution with an optional user is more correct, I'd suggest keeping it like this for now, and see how big of a complication it is for us. As I said, I'm open to changing it but I just want to wait till we see it's really a problem.

Reply:
I did the current user as optional at the beginning. But it was with a lot of noise to unwrap the current user. Additionally all our API will work only with logged in user after devs setup it. 
You can keep it. I recommend to think about that a little bit more

Reply:
As I said, I'm not against it, and you're probably right. I just try to keep it the most correct way as long as possible/convenient. Once it gets annoying, we change this.